test/generators/test_bimatrix_generators.jl

```diff
@@ -0,0 +1,211 @@
+import Combinatorics: binomial
+
+@testset "bimatrix_generators.jl" begin
+
+    @testset "blotto_game" begin
+        h = 3
+        t = 7
+        rho = 0.5
+
+        g = @inferred(blotto_game(h, t, rho))
+
+        @testset "test_nb_actions" begin
+            nb_actions = (factorial(t+h-1)/
+                          (factorial(h-1)*factorial(t)))
+
+            @test g.nums_actions == (nb_actions, nb_actions)
+        end
+
+        @testset "test_constant_diagonal" begin
+            for i=1:2
+                d = diag(g.players[i].payoff_array)
+                @test length(unique(d)) == 1
+            end
+        end
+
+        @testset "test_seed" begin
+            g1 = blotto_game(MersenneTwister(32), h, t, rho)
+            g2 = blotto_game(MersenneTwister(32), h, t, rho)
+            for i in 1:2
+                @test g1.players[i].payoff_array == g2.players[i].payoff_array
+            end
+        end
+
+        @testset "test_payoff_generation" begin
+            # WARNING: These results were computed by hand and depend on the
+            # structure of the function being tested. Altering the structure of
+            # the function will cause this test to fail.
```

Comment:
Is this warning still necessary after refactorizing?

Reply:
Yes, it think so. There may be further refactoring in the future.

Reply:
I think this is a good warning to have.